buildrunner/docker/image_info.py

```diff
@@ -0,0 +1,161 @@
+"""
+Copyright 2023 Adobe
+All Rights Reserved.
+
+NOTICE: Adobe permits you to use, modify, and distribute this file in accordance
+with the terms of the Adobe license agreement accompanying it.
+"""
+
+import logging
+import platform as python_platform
+from typing import Dict, List, Optional
+
+from pydantic import BaseModel
+
+
+LOGGER = logging.getLogger(__name__)
+
+# Maps from lower-case platform.system() result to platform prefix in docker
+PLATFORM_SYSTEM_MAPPINGS = {"darwin": "linux"}
+PLATFORM_MACHINE_MAPPINGS = {
+    "aarch64": "arm",
+    "x86_64": "amd",
+}
+
+
+class TaggedImageInfo(BaseModel):
+    # The image repo with no tag (e.g. repo/image-name)
+    repo: str
+    # All tags for this image
+    tags: List[str]
+
+    @property
+    def image_refs(self) -> List[str]:
+        """Returns a list of image names with tags"""
+        return [f"{self.repo}:{tag}" for tag in self.tags]
+
+    def __str__(self) -> str:
+        return f'{self.repo}:{",".join(self.tags)}'
+
+
+class BuiltTaggedImage(BaseModel):
+    # The image repo with no tag (e.g. repo/image-name)
+    repo: str
+    # The single tag for this image
+    tag: str
+    # Digest for the image
+    digest: str
+    # Platform for the image
+    platform: str
+
+    @property
+    def trunc_digest(self) -> str:
+        return self.digest.replace("sha256:", "")[:12]
+
+    @property
+    def image_ref(self) -> str:
+        """Returns an image name with the tag"""
+        return f"{self.repo}:{self.tag}"
+
+
+class BuiltImageInfo(BaseModel):
+    """
+    Contains information about images created during the build,
+    including the final tagged image name and any tags (once generated).
+    """
+
+    # A unique ID for this built image
+    id: str
+    # The built images by each platform
+    images_by_platform: Dict[str, BuiltTaggedImage] = {}
+    # The images that should be created from the built images referenced in each instance
+    tagged_images: List[TaggedImageInfo] = []
+
+    @property
+    def platforms(self) -> List[str]:
+        return list(self.images_by_platform.keys())
+
+    @property
+    def built_images(self) -> List[BuiltTaggedImage]:
+        return list(self.images_by_platform.values())
+
+    def image_for_platform(self, platform: str) -> Optional[BuiltTaggedImage]:
+        """Retrieves the built image for the given platform."""
+        if platform not in self.images_by_platform:
+            return None
+        return self.images_by_platform[platform]
```

Comment:
Could all this logic be replaced with `self.images_by_platform.get(platform)`, which would return None if there isn't a key for `platform`.

Reply:
🤦 yup, sure could. I must have had some more complex logic in here at some point... maybe... I updated it now though.